django-stubs/db/models/fields/related_descriptors.pyi

```diff
@@ -103,13 +104,17 @@ class ManyToManyDescriptor(ReverseManyToOneDescriptor, Generic[_M]):
 
     # 'field' here is 'rel.field'
     rel: ManyToManyRel  # type: ignore[assignment]
-    field: ManyToManyField[Any, _M]  # type: ignore[assignment]
+    field: ManyToManyField[_To, _Through]  # type: ignore[assignment]
     reverse: bool
     def __init__(self, rel: ManyToManyRel, reverse: bool = ...) -> None: ...
     @property
-    def through(self) -> type[_M]: ...
+    def through(self) -> type[_Through]: ...
     @property
-    def related_manager_cls(self) -> type[ManyRelatedManager[Any]]: ...  # type: ignore[override]
+    def related_manager_cls(self) -> type[ManyRelatedManager[_To]]: ...  # type: ignore[override]
+    @overload  # type: ignore[override]
+    def __get__(self, instance: None, cls: Any = ...) -> Self: ...
+    @overload
+    def __get__(self, instance: Model, cls: Any = ...) -> ManyRelatedManager[_To]: ...
```

Comment:
Just realised that the plugin needs to make the returned `ManyRelatedManager` here more accurate, so that it's possible to find custom queryset/manager methods.

Quite surprised we're missing tests for that actually.

It probably would've been easiest if `ManyRelatedManager` was another type argument, but I don't think it can. Since it's not possible to pass any dynamically generated managers as type arguments (ref #1699).

Reply:
I've adjusted so that we imitate Django's behaviour by having the plugin create a manager class with `ManyRelatedManager` and the `_To` model's default manager as bases.

Ref: https://github.com/django/django/blob/46df3ab244e1688bd186f0bfbfea6a354097a910/django/db/models/fields/related_descriptors.py#L1032